In [1]:
import pandas as pd
import numpy as np

# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)


0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


Zadanie 1a

In [3]:
dane_placowe = {'Anna': 5500, 'Jan':7200,'Ewa':4800,'Marek':9300, 'Kasia':6300}
pensje= pd.Series(dane_placowe)

Zadanie 1b

In [6]:
najbogatszy = pensje.idxmax()
print(f'Najwięcej zarabia: {najbogatszy}')
srednia= pensje.mean()
print(f'Srednia pensja:{srednia:.2f} zł')
powyzej_sredniej= pensje[pensje>srednia]
print('Osoby zarabiające powyżej sredniej')
print(powyzej_sredniej.index.tolist())
ponizej_6000 = (pensje < 6000).sum()
print(f'Liczba osób zarabiających poniżej 6000 :{ponizej_6000} ')

Najwięcej zarabia: Marek
Srednia pensja:6620.00 zł
Osoby zarabiające powyżej sredniej
['Jan', 'Marek']
Liczba osób zarabiających poniżej 6000 :2 


Zadanie 1c

In [7]:
podwyzka_10 = pensje *1.10
premia_500 = pensje + 500
nowe_pensje = (pensje *1.10)+500
print('\nNowe pensje (podwyżka 10% + 500 zł premii):')
print(nowe_pensje)
ranking = nowe_pensje.sort_values(ascending = False)
print('\nRanking pensji (od najwyższej):')
print(ranking)


Nowe pensje (podwyżka 10% + 500 zł premii):
Anna      6550.0
Jan       8420.0
Ewa       5780.0
Marek    10730.0
Kasia     7430.0
dtype: float64

Ranking pensji (od najwyższej):
Marek    10730.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


Cwiczenie 2

Zadanie2a

In [9]:
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
    'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
    'cena': [3500, 1800, 2500, 350, 2200],
    'stan_magazynowy': [45, 120, 200, 500, 75],
    'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})

In [10]:
print(f'Ksztalt:{sklep.shape}')
print(f'Nazwy kolumn: {sklep.columns.tolist()}')
print('\nTypy danych:')
print(sklep.dtypes)
print('\nPierwsze 3 wiersze:')
print(sklep.head(3))

Ksztalt:(5, 4)
Nazwy kolumn: ['produkt', 'cena', 'stan_magazynowy', 'kategoria']

Typy danych:
produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object

Pierwsze 3 wiersze:
    produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


Zadanie2b

In [11]:
sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']
sklep['cena_z_vat'] = sklep['cena'] *1.23
print ('Tabela z nowymi kolumnami:')
print(sklep)

Tabela z nowymi kolumnami:
     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


Zadanie2c

In [12]:
# Wczytaj dataset tips
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

In [13]:
print('Liczba wierszy i kolumn (tips):{tips.shape}')
print('\nOstatnie 5 wierszy')
print(tips.tail(5))
print('\n3losowe wiersze:')
print(tips.sample(3))
print('\nTypy danych w tips:')
print(tips.dtypes)

Liczba wierszy i kolumn (tips):{tips.shape}

Ostatnie 5 wierszy
     total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

3losowe wiersze:
     total_bill   tip     sex smoker   day    time  size
149        7.51  2.00    Male     No  Thur   Lunch     2
10        10.27  1.71    Male     No   Sun  Dinner     2
144       16.43  2.30  Female     No  Thur   Lunch     2

Typy danych w tips:
total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object


Ćwiczenie 3

Zadanie3a

In [14]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

In [16]:
print(penguins.head(10))
print(f'\nKsztalt tabeli:{penguins.shape}')
print('/nInformacja o strukturze i brakach')
print(penguins.info())

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

Zadanie3b

In [17]:
print('Statystyki kolumn numeryczbych:')
print(penguins.describe())
print('\nStatystyki wszystkich kolumn:')
print(penguins.describe(include='all'))
print(f'\nMediana masy ciała: {penguins['body_mass_g'].median()} g')

Statystyki kolumn numeryczbych:
       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

Statystyki wszystkich kolumn:
       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       1

Zadanie3c

In [19]:
print('Liczba pingwinów wg gatunku:')
print(penguins['species'].value_counts())
print('\nLiczba pingwinów na wyspach:')
print(penguins['island'].value_counts())
print('\nRozkład płci:')
print(penguins['sex'].value_counts())
print('\nUdział procentowy gatunów:')
print(penguins['species'].value_counts(normalize=True)*100)


Liczba pingwinów wg gatunku:
species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

Liczba pingwinów na wyspach:
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

Rozkład płci:
sex
MALE      168
FEMALE    165
Name: count, dtype: int64

Udział procentowy gatunów:
species
Adelie       44.186047
Gentoo       36.046512
Chinstrap    19.767442
Name: proportion, dtype: float64


Zadanie3d

In [21]:
braki = penguins.isna().sum()
print('Braki w kolumnach')
print(braki)
procent_brakow_sex = (braki['sex']/ len(penguins)) * 100
print(f'\nProcent brakujących danych w kolumnie "sex" :{procent_brakow_sex:.2f}%')


Braki w kolumnach
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

Procent brakujących danych w kolumnie "sex" :3.20%


"Dataset zawiera 344 pingwinów 3 gatunków: Adelie, Chinstrap i Gentoo"
"Najczęstszy gatunek to Adelie (152 osobników, 44%)"
"Brakuje danych w kolumnach dla 11 osobników"
"Średnia masa ciała to 4201 g (mediana: 4050 g)"
"Pingwiny mieszkają na 3 wyspach, najliczniej na Biscoe"

Ćwiczenie 4

Zadanie 4a

In [23]:
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')

# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

In [25]:
sredni_rachunek= tips['total_bill'].mean()
print(f'Sredni rachunek: {sredni_rachunek:.2f} $')

sredni_tip_pct = tips['tip_pct'].mean()
print(f'Sredni napiwek:{sredni_tip_pct:.1f} %')

najczestszy_dzien=tips['day'].value_counts().idxmax()
print(f'Dzień z najwięksą liczbą rachunków: {najczestszy_dzien}')

palacze_count = tips ['smoker'].value_counts()
print(f'rachunki od palaczy: {palacze_count}')

print('Statystyki dla kolumny tips')
print(tips['tip'].describe())
print(f'Mediana napiwku: {tips['tip'].median()}$')

procent_czasu = tips['time'].value_counts(normalize=True) * 100
print(f'Udział procentowy pór dnia: {procent_czasu}')
print(f'Procent rachunków "Dinner":{procent_czasu['Dinner']:.1f} %')

unikalne_grupy = tips['size'].nunique()
print(f'\nLiczba unikalnych rozmirów grup:{unikalne_grupy}')

Sredni rachunek: 19.79 $
Sredni napiwek:16.1 %
Dzień z najwięksą liczbą rachunków: Sat
rachunki od palaczy: smoker
No     151
Yes     93
Name: count, dtype: int64
Statystyki dla kolumny tips
count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64
Mediana napiwku: 2.9$
Udział procentowy pór dnia: time
Dinner    72.131148
Lunch     27.868852
Name: proportion, dtype: float64
Procent rachunków "Dinner":72.1 %

Liczba unikalnych rozmirów grup:6
